script.js

```diff
@@ -150,4 +150,19 @@ form.addEventListener('submit', ((e) => {
     errorMessage.style.color = 'red';
     e.preventDefault();
   }
-}));
\ No newline at end of file
+}));
+const data = {
+  name: '',
+  mail: '',
+  message: '',
+};
+const formItems = document.querySelectorAll('.form-container');
+const formData = Array.from(formItems);
+formData.forEach((form) => {
+  form.addEventListener('input', (e) => {
+    const letter = e.target.id;
+    data[letter] = e.target.value;
+    const dataMarker = JSON.stringify(data);
+    localStorage.setItem('data', dataMarker);
+  });
```

Comment:
- [ ] I can see the form information is stored in the local storage when the user inputs the data into the form. Which is a very great achievement 👏. I can say we are very close to completing the form data preserve. If you noticed, you are not having the part that can help you to retrieve data from local storage. Therefore I suggest that you modify your code to make sure that the contact form information is persistent to page reloads ☺️.